pinot-server/src/main/java/org/apache/pinot/server/api/resources/TablesResource.java

```diff
@@ -534,15 +547,18 @@ public ValidDocIdsBitmapResponse downloadValidDocIdsBitmap(
       if (validDocIdSnapshot == null) {
         String msg = String.format(
             "Found that validDocIds is missing while fetching validDocIds for table %s segment %s while "
-                + "reading the validDocIds with validDocIdType %s",
-            tableNameWithType, segmentDataManager.getSegmentName(), validDocIdsType);
+                + "reading the validDocIds with validDocIdType %s", tableNameWithType,
+            segmentDataManager.getSegmentName(), validDocIdsType);
         LOGGER.warn(msg);
         throw new WebApplicationException(msg, Response.Status.NOT_FOUND);
       }
-
       byte[] validDocIdsBytes = RoaringBitmapUtils.serialize(validDocIdSnapshot);
       return new ValidDocIdsBitmapResponse(segmentName, indexSegment.getSegmentMetadata().getCrc(),
-          finalValidDocIdsType, validDocIdsBytes);
+          finalValidDocIdsType, validDocIdsBytes, _serverInstance.getInstanceDataManager().getInstanceId(),
+          serverStatus);
+    } catch (Exception e) {
+      LOGGER.error("Failed to get validDocIds for table {}: {}", tableNameWithType, e.getMessage());
+      return null;
```

Comment:
why return null ? if there is an expected exception we should catch and wrap it in WebApplicationException. For unexpected exceptions it will be better to throw it back.